language/move-model/src/spec_translator.rs

```diff
@@ -606,33 +605,20 @@ impl<'a, 'b, T: ExpGenerator<'a>> ExpRewriterFunctions for SpecTranslator<'a, 'b
 
     fn rewrite_temporary(&mut self, id: NodeId, idx: TempIndex) -> Option<Exp> {
         // Compute the effective index.
-        let local_type = if idx < self.fun_env.get_parameter_count() {
-            // if the idx is a function argument, get its original type
-            self.fun_env.get_local_type(idx)
-        } else {
-            // otherwise, get type from the builder
-            self.builder.get_local_type(idx)
-        };
-        let is_mut = local_type.is_mutable_reference();
-        let effective_idx = if self.in_old || self.in_post_state && !is_mut {
+        let mut effective_idx = self.apply_param_substitution(idx);
```

Comment:
Nice find of simplification.

At some time in the past a parameter could have a different type in the fun_env (`&mut T`) as in the fun_target (`T`), as a result of memory model transformation. But this was removed, so your simplification looks legit.